src/core/controls.js

```diff
@@ -0,0 +1,177 @@
+var component = require('./component');
+var systems = require('./system');
+var utils = require('../utils/');
+
+// Keep track of registered controls.
+var controls = module.exports.controls = {
+  movementControls: {},
+  rotationControls: {}
+};
+
+/**
+ * Initializes the A-Frame controls system.
+ */
+systems.registerSystem('controls', {
+  init: function () {
+    // Names of enabled movement/rotation controls, in order of precedence.
+    this.enabledMovementControls = [];
+    this.enabledRotationControls = [];
+
+    // Controls components registered to update object position/rotation.
+    this.movementControls = controls.movementControls;
+    this.rotationControls = controls.rotationControls;
+  },
+
+  /**
+   * Enables the given component names, and sets the given order as their precedence.
+   * @param {Array<string>} controls
+   */
+  enableMovementControls: function (controls) {
+    this.enabledMovementControls = controls;
+  },
+
+  /**
+   * Enables the given component names, and sets the given order as their precedence.
+   * @param {Array<string>} controls
+   */
+  enableRotationControls: function (controls) {
+    this.enabledRotationControls = controls;
+  },
+
+  /**
```

Comment:
I think what a system could provide for now, is just a list of registered rotation + movement controls. So that controls component can check for validity.